app/Database/Models/AuthToken.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * @author    MyBB Group
+ * @version   2.0.0
+ * @package   mybb/core
+ * @license   http://www.mybb.com/licenses/bsd3 BSD-3
+ */
+
+namespace MyBB\Core\Database\Models;
+
+use DateTime;
+use MyBB\Core\Exceptions\InvalidConfirmationTokenException;
+use MyBB\Core\Database\AbstractModel;
+
+/**
+ * @todo document database columns with @property
+ */
+class AuthToken extends AbstractModel
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected $table = 'auth_tokens';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $dates = ['created_at'];
+
+    /**
+     * Use a custom primary key for this model.
+     *
+     * @var bool
+     */
+    public $incrementing = false;
+
+    /**
+     * Generate an email token for the specified user.
+     *
+     * @param string $email
+     *
+     * @return static
+     */
+    public static function generate($payload)
```

Comment:
I don't like this similar logic being duplicated both here and in the repository - it would be better if it was just here and the repository just called this static method IMO.